spec.html

```diff
@@ -28485,47 +28485,43 @@ <h1>String.prototype.padEnd ( _maxLength_ [ , _fillString_ ] )</h1>
-        </emu-note>
-        <emu-note>
-          <p>The optional second argument _fillString_ defaults to *" "* (the String value consisting of the code unit 0x0020 SPACE).</p>
-        </emu-note>
       </emu-clause>
 
       <emu-clause id="sec-string.prototype.padstart">
         <h1>String.prototype.padStart ( _maxLength_ [ , _fillString_ ] )</h1>
         <p>When the `padStart` method is called, the following steps are taken:</p>
         <emu-alg>
           1. Let _O_ be ? RequireObjectCoercible(*this* value).
-          1. Let _S_ be ? ToString(_O_).
-          1. Let _intMaxLength_ be ? ToLength(_maxLength_).
-          1. Let _stringLength_ be the length of _S_.
-          1. If _intMaxLength_ is not greater than _stringLength_, return _S_.
-          1. If _fillString_ is *undefined*, let _filler_ be the String value consisting solely of the code unit 0x0020 (SPACE).
-          1. Else, let _filler_ be ? ToString(_fillString_).
-          1. If _filler_ is the empty String, return _S_.
-          1. Let _fillLen_ be _intMaxLength_ - _stringLength_.
-          1. Let _truncatedStringFiller_ be the String value consisting of repeated concatenations of _filler_ truncated to length _fillLen_.
-          1. Return the string-concatenation of _truncatedStringFiller_ and _S_.
+          1. Return ? StringPad(_O_, _maxLength_, _fillString_, *"start"*).
         </emu-alg>
-        <emu-note>
-          <p>The first argument _maxLength_ will be clamped such that it can be no smaller than the length of the *this* value.</p>
-        </emu-note>
-        <emu-note>
-          <p>The optional second argument _fillString_ defaults to *" "* (the String value consisting of the code unit 0x0020 SPACE).</p>
-        </emu-note>
+
+        <emu-clause id="sec-stringpad" aoid="StringPad">
+          <h1>Runtime Semantics: StringPad ( _O_, _maxLength_, _fillString_, _placement_ )</h1>
+          <p>When the abstract operation StringPad is called with arguments _O_, _maxLength_, _fillString_, and _placement_, the following steps are taken:</p>
+          <emu-alg>
+            1. Let _S_ be ? ToString(_O_).
+            1. Let _intMaxLength_ be ? ToLength(_maxLength_).
+            1. Let _stringLength_ be the length of _S_.
+            1. If _intMaxLength_ is not greater than _stringLength_, return _S_.
+            1. If _fillString_ is *undefined*, let _filler_ be the String value consisting solely of the code unit 0x0020 (SPACE).
+            1. Else, let _filler_ be ? ToString(_fillString_).
+            1. If _filler_ is the empty String, return _S_.
+            1. Let _fillLen_ be _intMaxLength_ - _stringLength_.
+            1. Let _truncatedStringFiller_ be the String value consisting of repeated concatenations of _filler_ truncated to length _fillLen_.
+            1. If _placement_ is *"end"*, return the string-concatenation of _S_ and _truncatedStringFiller_.
+            1. Else,
+              1. Assert: _placement_ is *"start"*.
```

Comment:
let's maybe assert that `_placement_` is start or end at the top of the operation, before the observable things happen? seems a bit more clear to me.

Reply:
Sure; done.